sycl/doc/extensions/Assert/opencl.md

```diff
@@ -0,0 +1,22 @@
+# Overview
+
+This extension enables detection of assert failure of kernel.
```

Comment:
Suggest following style of other OpenCL extensions, for example this one:
https://www.khronos.org/registry/OpenCL/extensions/intel/cl_intel_command_queue_families.html

List of all extensions: https://www.khronos.org/registry/OpenCL/

Reply:
It's not clear to me if this document is an extension specification or a design document (or both).

We should have an extension specification for this feature because it adds a new device-side function `assert()` and it adds a new exception type.  We've decided to use a different format for SYCL 2020 extensions, so I'd suggest following (https://gitlab.devtools.intel.com/SYCL/extensions/-/merge_requests/74) as a model.  (Sorry for the internal link, this will move to the public repo soon.)

It may make sense to have a separate design document since the design details will not be as interesting to users once it is implemented.  It looks like extension specifications normally go in "sycl/doc/extensions" whereas design documents go in "sycl/doc".

Reply:
This specific comment is about extension for OpenCL, not for SYCL. But, right, we need a formal extension document for SYCL as well.

Reply:
It's not clear to me if this document is an extension specification or a design document (or both).

We should have an extension specification because this proposal adds a new device-side function `assert()` and a new exception type.  For SYCL 2020, we've decided to follow a new format.  I'll send you a template separately via email.

We should also have a design document, and it may make sense to have a separate document since customers won't be interested in the design details once it's implemented.

It looks like we put extension documents in "sycl/doc/extensions" and we put design documents in "sycl/doc".

Reply:
@gmlueck , the design document is added within this patch: `sycl/doc/Assert.md` - https://github.com/intel/llvm/pull/3461/files#diff-0f73edd3d2f442213cd81f706edb7a277172d98f0fd029333888d5da2977ca2f